msm-sitemap.php

```diff
@@ -570,24 +571,50 @@ public static function create_post_type() {
 			);
 	}
 
+	/**
+	 * Record deleted posts in an option to be used by get_last_modified_posts.
+	 */
+	public static function record_sitemap_to_rebuild( $post_id ) {
+		$deleted_posts = get_option( 'msm_sitemap_deleted_posts', array() );
+		$post = get_post( $post_id );
+		if ( ! $post ) {
+			return;
+		}
+
+		if ( in_array( $post->post_type, self::get_supported_post_types() ) ) {
+			$deleted_posts[] = array(
+				'ID' => (string) $post->ID,
+				'post_date' => $post->post_date,
+				);
+			update_option( 'msm_sitemap_deleted_posts', $deleted_posts, false );
+		}
+	}
+
 	/**
 	 * Get posts modified within the last hour
 	 * @return object[] modified posts
 	 */
-	public static function get_last_modified_posts() {
+	public static function get_last_modified_posts( $start_time = null) {
 		global $wpdb;
 
+		if ( ! $start_time ) {
+			$start_time = current_time( 'timestamp', 1 );
+		}
+
 		$sitemap_last_run = get_option( 'msm_sitemap_update_last_run', false );
 
-		$date = date( 'Y-m-d H:i:s', ( current_time( 'timestamp', 1 ) - 3600 ) ); // posts changed within the last hour
+		$date = date( 'Y-m-d H:i:s', ( $start_time - 3600 ) ); // posts changed within the last hour
```

Comment:
I think we should add a comment to why we're adding the start time. Either inline or with the doc block, whatever makes most sense.

Reply:
Added in https://github.com/Automattic/msm-sitemap/pull/135/commits/414ce0ef7cc513adbcfe047e34d3d8a436461635